paymentsheet/src/test/java/com/stripe/android/paymentsheet/CustomerSessionPaymentSheetActivityTest.kt

```diff
@@ -0,0 +1,422 @@
+
+            composeTestRule.onEditScreenRemoveButton().assertDoesNotExist()
+        }
+    }
+
+    private fun activityScenario(
+        paymentMethods: List<PaymentMethod>,
+        isPaymentMethodRemoveEnabled: Boolean,
+        allowsRemovalOfLastSavedPaymentMethod: Boolean,
+    ): InjectableActivityScenario<PaymentSheetActivity> {
+        return injectableActivityScenario<PaymentSheetActivity> {
+            injectActivity {
+                val arguments = PaymentSheetContractV2.Args.fromIntent(intent)
+                    ?: throw IllegalStateException("Should initialize intent with PaymentSheet arguments!")
+
+                viewModelFactory = viewModelFactoryFor(
+                    createViewModel(
+                        arguments,
+                        paymentMethods,
+                        isPaymentMethodRemoveEnabled
+                    )
+                )
+            }
+        }.launch(
+            PaymentSheetContractV2().createIntent(
+                ApplicationProvider.getApplicationContext(),
+                PaymentSheetContractV2.Args(
+                    initializationMode = PaymentSheet.InitializationMode.PaymentIntent(
+                        clientSecret = "pi_1234_secret_5678",
+                    ),
+                    config = PaymentSheet.Configuration(
+                        merchantDisplayName = "Merchant, Inc.",
+                        customer = PaymentSheet.CustomerConfiguration.createWithCustomerSession(
+                            id = "cus_1",
+                            clientSecret = "cuss_1",
+                        ),
+                        allowsRemovalOfLastSavedPaymentMethod = allowsRemovalOfLastSavedPaymentMethod,
+                        preferredNetworks = listOf(CardBrand.CartesBancaires, CardBrand.Visa),
+                    ),
+                    statusBarColor = PaymentSheetFixtures.STATUS_BAR_COLOR,
+                )
+            )
+        )
+    }
+
+    private fun createViewModel(
+        arguments: PaymentSheetContractV2.Args,
+        paymentMethods: List<PaymentMethod>,
+        isPaymentMethodRemoveEnabled: Boolean,
+    ): PaymentSheetViewModel = runBlocking {
+        TestViewModelFactory.create(
+            linkConfigurationCoordinator = mock<LinkConfigurationCoordinator>().stub {
+                onBlocking { getAccountStatusFlow(any()) }.thenReturn(flowOf(AccountStatus.SignedOut))
+                on { emailFlow } doReturn stateFlowOf("email@email.com")
+            },
+        ) { linkHandler, linkInteractor, savedStateHandle ->
+            val application = ApplicationProvider.getApplicationContext<Application>()
+            val customer = PaymentSheetFixtures.EMPTY_CUSTOMER_STATE
+
+            PaymentSheetViewModel(
+                application = application,
+                args = arguments,
+                eventReporter = mock(),
+                paymentSheetLoader = FakePaymentSheetLoader(
```

Comment:
Should we replace this with `NetworkRule`? Seems nice to have a more "real" network stack.